apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/ProjectsResourceTest.java

```diff
@@ -858,6 +857,69 @@ void getProjects__whenSortingProjectsByLastTrace__thenReturnProjectSorted(Direct
+            String apiKey = UUID.randomUUID().toString();
+            String workspaceId = UUID.randomUUID().toString();
+
+            mockTargetWorkspace(apiKey, workspaceName, workspaceId);
+
+            List<Project> withTraceProjects = PodamFactoryUtils.manufacturePojoList(factory, Project.class);
+
+            withTraceProjects = withTraceProjects.stream().map(project -> {
+                UUID projectId = createProject(project, apiKey, workspaceName);
+                List<UUID> traceIds = IntStream.range(0, 5)
+                        .mapToObj(i -> createCreateTrace(project.name(), apiKey, workspaceName))
+                        .toList();
+
+                Trace trace = getTrace(traceIds.getLast(), apiKey, workspaceName);
+                return project.toBuilder()
+                        .id(projectId)
+                        .lastUpdatedTraceAt(trace.lastUpdatedAt()).build();
+            }).toList();
+
+            // add a project with no traces
+            List<Project> noTraceProjects = PodamFactoryUtils.manufacturePojoList(factory, Project.class)
+                    .stream().map(project -> {
+                        UUID projectId = createProject(project, apiKey, workspaceName);
+                        return project.toBuilder().id(projectId).build();
+                    }).toList();
+            List<Project> allProjects = Stream.concat(withTraceProjects.stream(), noTraceProjects.stream()).toList();
+
+            var sorting = List.of(SortingField.builder()
+                    .field(SortableFields.LAST_UPDATED_TRACE_AT)
+                    .direction(request)
+                    .build());
+
+            var actualResponse = client.target(URL_TEMPLATE.formatted(baseURI))
+                    .queryParam("size", allProjects.size())
+                    .queryParam("sorting", URLEncoder.encode(JsonUtils.writeValueAsString(sorting),
+                            StandardCharsets.UTF_8))
+                    .request()
+                    .header(HttpHeaders.AUTHORIZATION, apiKey)
+                    .header(WORKSPACE_HEADER, workspaceName)
+                    .get();
+
+            var actualEntity = actualResponse.readEntity(Project.ProjectPage.class);
+
+            assertThat(actualResponse.getStatusInfo().getStatusCode()).isEqualTo(200);
+            assertThat(actualEntity.size()).isEqualTo(allProjects.size());
+            assertThat(actualEntity.total()).isEqualTo(allProjects.size());
+            assertThat(actualEntity.page()).isEqualTo(1);
+
+            var allExpectedProjects = Stream.concat(withTraceProjects.stream(), noTraceProjects.stream()).toList();
+            if (expected == Direction.DESC) {
+                allExpectedProjects = allExpectedProjects.reversed();
+            }
+
+            assertThat(actualEntity.content()).usingRecursiveFieldByFieldElementComparatorIgnoringFields(IGNORED_FIELDS)
+                    .containsExactlyElementsOf(allExpectedProjects);
```

Comment:
This part can probably be extracted in a private method as it's similar to all tests